{
  "url": "https://api.github.com/repos/quarkusio/quarkus/issues/29589",
  "repository_url": "https://api.github.com/repos/quarkusio/quarkus",
  "labels_url": "https://api.github.com/repos/quarkusio/quarkus/issues/29589/labels{/name}",
  "comments_url": "https://api.github.com/repos/quarkusio/quarkus/issues/29589/comments",
  "events_url": "https://api.github.com/repos/quarkusio/quarkus/issues/29589/events",
  "html_url": "https://github.com/quarkusio/quarkus/issues/29589",
  "id": 1469985286,
  "node_id": "I_kwDOCFbutM5XnjIG",
  "number": 29589,
  "title": "Incorrect quarkus.http.ssl.certificate.key-store-password ends with misleading NSFE for the keystore",
  "labels": [
    {
      "id": 985346620,
      "node_id": "MDU6TGFiZWw5ODUzNDY2MjA=",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/kind/bug",
      "name": "kind/bug",
      "color": "d73a4a",
      "default": false,
      "description": "Something isn't working"
    },
    {
      "id": 1273039603,
      "node_id": "MDU6TGFiZWwxMjczMDM5NjAz",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/area/devmode",
      "name": "area/devmode",
      "color": "0366d6",
      "default": false,
      "description": ""
    },
    {
      "id": 1282139374,
      "node_id": "MDU6TGFiZWwxMjgyMTM5Mzc0",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/area/security",
      "name": "area/security",
      "color": "0366d6",
      "default": false,
      "description": ""
    }
  ],
  "state": "open",
  "locked": false,
  "milestone": null,
  "comments": 2,
  "created_at": "2022-11-30T17:25:17Z",
  "updated_at": "2022-12-05T13:32:36Z",
  "closed_at": null,
  "active_lock_reason": null,
  "body": "### Describe the bug\n\nIncorrect `quarkus.http.ssl.certificate.key-store-password` ends with misleading NSFE for the keystore\r\n\r\nWhen the app starts (in dev mode) I receive 5 errors with stacktraces, last two (stacktrace nr. 4 and 5) have `Caused by: java.nio.file.NoSuchFileException: META-INF/resources/server.keystore`\r\n\r\nStacktraces nr. 2 and 3 have the info `Caused by: java.io.IOException: keystore password was incorrect` so patient user can find the root cause.\r\nBut my experience is that people often look at the last message/stacktrace and do not check all the the above messages.\r\n\r\n**AI:** I think the case with incorrect password could have better user experience and less stacktraces.\r\n\r\nThe last stacktrace:\r\n```\r\n2022-11-30 18:15:20,083 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.nio.file.NoSuchFileException: META-INF/resources/server.keystore\r\n\tat io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:281)\r\n\tat io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:61)\r\n\tat io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:1139)\r\n\tat io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:175)\r\n\tat io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)\r\n\tat io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)\r\n\tat io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)\r\n\tat io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)\r\n\tat io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)\r\n\tat io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)\r\nCaused by: java.nio.file.NoSuchFileException: META-INF/resources/server.keystore\r\n\tat java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)\r\n\tat java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)\r\n\tat java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)\r\n\tat java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)\r\n\tat java.base/java.nio.file.Files.newByteChannel(Files.java:380)\r\n\tat java.base/java.nio.file.Files.newByteChannel(Files.java:432)\r\n\tat java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)\r\n\tat java.base/java.nio.file.Files.newInputStream(Files.java:160)\r\n\tat io.quarkus.vertx.http.runtime.VertxHttpRecorder.getFileContent(VertxHttpRecorder.java:971)\r\n\tat io.quarkus.vertx.http.runtime.VertxHttpRecorder.createKeyStoreOptions(VertxHttpRecorder.java:951)\r\n\tat io.quarkus.vertx.http.runtime.VertxHttpRecorder.createSslOptions(VertxHttpRecorder.java:860)\r\n\tat io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:657)\r\n\tat io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:274)\r\n\t... 9 more\r\n```\n\n### Expected behavior\n\nNot so many stacktraces + clear message about incorrect password at the end\n\n### Actual behavior\n\nLong stack trace with misleading NSFE for the keystore\n\n### How to Reproduce?\n\n - https://github.com/rsvoboda/code-with-quarkus\r\n - `mvn quarkus:dev -Dquarkus.http.ssl.certificate.key-store-password=bleh`\n\n### Output of `uname -a` or `ver`\n\nmacOS\n\n### Output of `java -version`\n\nJava 17\n\n### GraalVM version (if different from Java)\n\n_No response_\n\n### Quarkus version or git rev\n\n2.14.2.Final\n\n### Build tool (ie. output of `mvnw --version` or `gradlew --version`)\n\n_No response_\n\n### Additional information\n\n_No response_",
  "reactions": {
    "url": "https://api.github.com/repos/quarkusio/quarkus/issues/29589/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/quarkusio/quarkus/issues/29589/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
